pkg/cli/image/mirror/mirror.go

```diff
@@ -151,8 +157,8 @@ func NewCmdMirrorImage(streams genericclioptions.IOStreams) *cobra.Command {
 		Long:    mirrorDesc,
 		Example: mirrorExample,
 		Run: func(c *cobra.Command, args []string) {
-			kcmdutil.CheckErr(o.Complete(c, args))
 			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.Complete(c, args))
```

Comment:
This order doesn't look right? Why would `Validate` be before `Complete`?

Reply:
it's because in FilterOptions.Validate, the option OSFilter is set, and that has to be set only after Complete (unless we fix that and rewrite both Validate and Complete, but going down that path I saw some snags so opted for this, for now, instead)

Reply:
although, I think _Validate_ shouldn't _set_ any option, that should be done in _Complete_ - but the order in this particular case is tricky, will take some time to detangle that (I was stuck on it for a bit so left it as/is for now)

Reply:
The only thing Validate is there to catch is the conflicting `--keep-manifest-list=true` with `--filter-by-os=not/wildcard`